Update to m137

pblazej opened the pull request:
I changed my strategy and merged old → new, not as clear, but the amount of conflicts in the Obj-C++ part (especially around prefixing) made it unmanageable.

## Status

### Apple
- 🟢 iOS - works
- 🟢 macOS - works
- 🟢 Catalyst - works
- 🟡 tvOS - runs
- 🟡 xrOS - runs, requires 2.0 ⚠️ 
```
ld64.lld: error: ../../third_party/llvm-build/Release+Asserts/lib/clang/21/lib/darwin/libclang_rt.xros.a(addtf3.c.o) has version 2.2.0, which is newer than target minimum of 1.1.0
```

### Android
- 🟢 works

### Flutter
- 🟢 works

### Tests
- unknown

## Other PRs
- https://github.com/webrtc-sdk/build/pull/3
- https://github.com/webrtc-sdk/webrtc-build/pull/32

perkj committed:
Revert "Ignore allocated bitrate during initial exponential BWE."

This reverts commit 33cc83595ae7dd144c57c614fb62d286d9d7bf68.

Reason for revert: Perf bots showed that this cl cause a change in metrics. It looks like it is for the better, but we want this to be behind a field trial.

Original change's description:
> Ignore allocated bitrate during initial exponential BWE.
>
> The reason why we want to do this is  because audio can allocate a needed bitrate before video when starting a call, which may lead to a race between the first probe result and updating the allocated bitrate.
> That is the, initial probe will try to probe up to the max configured bitrate.
>
> ProbeController::SetFirstProbeToMaxBitrate will allow the first probe to
> continue up to the max configured bitrate, regardless of of the max
> allocated bitrate.
>
> Bug: webrtc:14928
> Change-Id: I6e0ae90e21a78466527f3464951e6033dc846470
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/346760
> Reviewed-by: Diep Bui <diepbp@webrtc.org>
> Commit-Queue: Per Kjellander <perkj@webrtc.org>
> Reviewed-by: Erik Språng <sprang@webrtc.org>
> Reviewed-by: Per Kjellander <perkj@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#42049}

(cherry picked from commit 501c4f37bfee47b26999ee291c5355ad64554df7)

Bug: chromium:335337923,webrtc:14928
Change-Id: I56ba58560b6857b6069552c02df822691f7af64d
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/347622
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Per Kjellander <perkj@webrtc.org>
Reviewed-by: Diep Bui <diepbp@webrtc.org>
Owners-Override: Per Kjellander <perkj@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#42081}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/348722
Reviewed-by: Erik Språng <sprang@webrtc.org>
Cr-Commit-Position: refs/branch-heads/6422@{#2}
Cr-Branched-From: b831eb816ef847d09d446ef4168e36b13af163f8-refs/heads/main@{#42072}

fippo committed:
[m125] webrtc stats: fix video remote-outbound-rtp timestamp

which had a 70 year offset (i.e. 2094 instead of 2024) which broke
the webrtc-internal stats graphs. A similar adjustment is done
for audio in audio/channel_receive.cc

BUG=webrtc:12529,chromium:336222282

(cherry picked from commit 77313800c72fa1e33c30e952800e4157e9ad44a4)

Change-Id: I0ce43cc8b451185bc056cf9e54757ef22d006c99
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/347780
Reviewed-by: Florent Castelli <orphis@webrtc.org>
Commit-Queue: Philipp Hancke <phancke@meta.com>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#42114}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/348702
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Cr-Commit-Position: refs/branch-heads/6422@{#1}
Cr-Branched-From: b831eb816ef847d09d446ef4168e36b13af163f8-refs/heads/main@{#42072}

cloudwebrtc committed:
Update to m125. (#119)

Use M125 as the latest version and migrate historical patches to m125

Patches Group:

## 1. Update README.md
https://github.com/webrtc-sdk/webrtc/pull/119/commits/b6c65fc7f0e81ed305e5578467b7198833665352
* Add Apache-2.0 license and some note to README.md. (#9)
* Updated readme detailing changes from original (#42)
* Adding membrane framework (#51)
* Updated readme (#83)

## 2. Audio Device Optimization
https://github.com/webrtc-sdk/webrtc/pull/119/commits/7454824ea6481cae2f70ad58fb1e1eb5d3ecdc88
* allow listen-only mode in AudioUnit, adjust when category changes
(https://github.com/webrtc-sdk/webrtc/pull/2)
* release mic when category changes
(https://github.com/webrtc-sdk/webrtc/pull/5)
* Change defaults to iOS defaults
(https://github.com/webrtc-sdk/webrtc/pull/7)
* Sync audio session config
(https://github.com/webrtc-sdk/webrtc/pull/8)
* feat: support bypass voice processing for iOS.
(https://github.com/webrtc-sdk/webrtc/pull/15)
* Remove MacBookPro audio pan right code
(https://github.com/webrtc-sdk/webrtc/pull/22)
* fix: Fix can't open mic alone when built-in AEC is enabled.
(https://github.com/webrtc-sdk/webrtc/pull/29)
* feat: add audio device changes detect for windows.
(https://github.com/webrtc-sdk/webrtc/pull/41)
* fix Linux compile (https://github.com/webrtc-sdk/webrtc/pull/47)
* AudioUnit: Don't rely on category switch for mic indicator to turn off
(https://github.com/webrtc-sdk/webrtc/pull/52)
* Stop recording on mute (turn off mic indicator)
(https://github.com/webrtc-sdk/webrtc/pull/55)
* Cherry pick audio selection from m97 release
(https://github.com/webrtc-sdk/webrtc/pull/35)
* [Mac] Allow audio device selection
(https://github.com/webrtc-sdk/webrtc/pull/21)
* RTCAudioDeviceModule.outputDevice / inputDevice getter and setter
(https://github.com/webrtc-sdk/webrtc/pull/80)
* Allow custom audio processing by exposing AudioProcessingModule
(https://github.com/webrtc-sdk/webrtc/pull/85)
* Expose audio sample buffers for Android
(https://github.com/webrtc-sdk/webrtc/pull/89)
* feat: add external audio processor for android.
(https://github.com/webrtc-sdk/webrtc/pull/103)
* android: make audio output attributes modifiable
(https://github.com/webrtc-sdk/webrtc/pull/118)
* Fix external audio processor sample rate calculation
(https://github.com/webrtc-sdk/webrtc/pull/108)
* Expose remote audio sample buffers on RTCAudioTrack
(https://github.com/webrtc-sdk/webrtc/pull/84)
* Fix memory leak when creating audio CMSampleBuffer
https://github.com/webrtc-sdk/webrtc/pull/86

## 3. Simulcast/SVC support for iOS/Android.
https://github.com/webrtc-sdk/webrtc/pull/119/commits/b0b9fe93b1b4f0f8cdd80919a05f999124ff8bce
    
- Simulcast support for iOS SDK (#4)
- Support for simulcast in Android SDK (#3)
- include simulcast headers for mac also (#10)
- Fix simulcast using hardware encoder on Android (#48)
- Add scalabilityMode support for AV1/VP9. (#90)

## 4. Android improvements.
https://github.com/webrtc-sdk/webrtc/pull/119/commits/9aaaab5a20fec2c84616ca709ac9c1aae6de9225
- Start/Stop receiving stream method for VideoTrack (#25)
- Properly remove observer upon deconstruction (#26)
- feat: Expose setCodecPreferences/getCapabilities for android. (#61)
- fix: add WrappedVideoDecoderFactory.java. (#74)

## 5. Darwin improvements
https://github.com/webrtc-sdk/webrtc/pull/119/commits/a13ea178e46ee2ab222f1553c9164b6556e1abd8
- [Mac/iOS] feat: Add RTCYUVHelper for darwin. (#28)
- Cross-platform `RTCMTLVideoView` for both iOS / macOS (#40)
- rotationOverride should not be assign (#44)
- [ObjC] Expose properties / methods required for AV1 codec support
(#60)
- Workaround: Render PixelBuffer in RTCMTLVideoView (#58)
- Improve iOS/macOS H264 encoder (#70)
- fix: fix video encoder not resuming correctly upon foregrounding
(#75).
- add PrivacyInfo.xcprivacy to darwin frameworks. (#112)
- Add NSPrivacyCollectedDataTypes key to xcprivacy file (#114)
- Thread-safe `RTCInitFieldTrialDictionary` (#116)
- Set RTCCameraVideoCapturer initial zoom factor (#121)
- Unlock configuration before starting capture session (#122)

## 6. Desktop Capture for macOS.
https://github.com/webrtc-sdk/webrtc/pull/119/commits/841d78f0631aff91590c1493909c4f3fccd8fafe
- [Mac] feat: Support screen capture for macOS. (#24) (#36)
- fix: Get thumbnails asynchronously. (#37)
- fix: Use CVPixelBuffer to build DesktopCapture Frame, fix the crash
caused by non-CVPixelBuffer frame in RTCVideoEncoderH264 that cannot be
cropped. (#63)
- Fix the crash when setting the fps of the virtual camera. (#62)

## 7. Frame Cryptor Support.
https://github.com/webrtc-sdk/webrtc/pull/119/commits/fc08745d9c5218aed45829942a3590c97c19c76a
- feat: Frame Cryptor (aes gcm/cbc). (#54)
- feat: key ratchet/derive. (#66)
- fix: skip invalid key when decryption failed. (#81)
- Improve e2ee, add setSharedKey to KeyProvider. (#88)
- add failure tolerance for framecryptor. (#91)
- fix h264 freeze. (#93)
- Fix/send frame cryptor events from signaling thread (#95)
- more improvements for E2EE. (#96)
- remove too verbose logs (#107)
- Add key ring size to keyProviderOptions. (#109)

## 8. Other improvements.
https://github.com/webrtc-sdk/webrtc/pull/119/commits/eed6c8a1bff6c0e16a850be07597e59ed8a68d7e
- Added yuv_helper (#57)
- ABGRToI420, ARGBToI420 & ARGBToRGB24 (#65)
- more yuv wrappers (#87)
- Fix naming for yuv helper (#113)
- Fix missing `RTC_OBJC_TYPE` macros (#100)

---------

Co-authored-by: Hiroshi Horie <548776+hiroshihorie@users.noreply.github.com>
Co-authored-by: David Zhao <dz@livekit.io>
Co-authored-by: davidliu <davidliu@deviange.net>
Co-authored-by: Angelika Serwa <angelika.serwa@gmail.com>
Co-authored-by: Théo Monnom <theo.monnom@outlook.com>

hiroshihorie committed:
Fix missing RTC_OBJC_TYPE macros

hiroshihorie committed:
Fix missing headers and Metal linking

hiroshihorie committed:
Fix Mac Catalyst `RTCCameraVideoCapturer` rotation (#126)

hiroshihorie committed:
Fix set frame transformer (#125)

davidliu committed:
Fix webrtc_voice_engine not notifying mute change (#128)

Looks like this line was missed during the m125 update.


https://github.com/webrtc-sdk/webrtc/commit/272127d457ab48e36241e82549870405864851f6#diff-56f5e0c459b287281ef3b0431d3f4129e8e4be4c6955d845bcb22210f08b7ba5R2289

Adding it back in so that mic is properly released when muted.

davidliu committed:
android: Allow for skipping checking the audio playstate if needed (#129)

Pausing/stopping the audio track can lead to a race condition against
the AudioTrackThread due to this assert. Normally this is fine since
directly pausing/stopping isn't possible, but user is using reflection
to workaround another audio issue (muted participants still have a
sending audio stream which keeps the audio alive, affecting global sound
if in the background).

Not a full fix, as would like to manually control the audio track
directly (needs a bigger fix to handle proper synchronization before
allowing public access), but this will work through reflection (user
takes responsibility for usage).

hiroshihorie committed:
Allow to pass in capture session to RTCCameraVideoCapturer (#132)

Expose initializers to pass in capture session to RTCCameraVideoCapturer
so we can use AVCaptureMultiCamSession etc to capture front and back
simultaneously for iOS.

davidliu committed:
Fix NetworkMonitor race condition when dispatching native observers (#135)

There is a race condition in NetworkMonitor where native observers may
be removed concurrently with a notification being dispatched, leading to
a dangling pointer dereference (trying to dispatch an observer that was
already removed and destroyed), and from there a crash with access
violation.

By ensuring dispatching to native observers is done within the
synchronization lock that guards additions/removals of native observers
protects against this race condition. Since native observers callbacks
are posted to the networking thread in the C++ side anyway, there should
be no risk of deadlock/starvation due to long-running observers.

Bug: webrtc:15837
Change-Id: Id2b788f102dbd25de76ceed434c4cd68aa9a569e
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/338643
Reviewed-by: Taylor Brandstetter <deadbeef@webrtc.org>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#42256}

Co-authored-by: Guy Hershenbaum <hershi@meta.com>

cloudwebrtc committed:
Support for Vision Pro (#131)

TODO:
- [x]  fix compile for RTCCameraVideoCapturer
- [ ]  fix RTCMTLRenderer ?

---------

Co-authored-by: Hiroshi Horie <548776+hiroshihorie@users.noreply.github.com>

hiroshihorie committed:
Multicam support (#137)

TODO: 
- [x] Return `.systemPreferredCamera` for devices (visionOS only).
- [x] Use `AVCaptureMultiCamSession` only if `isMultiCamSupported` is
true.
- [x] Silence statusBarOrientation warning.

---------

Co-authored-by: duanweiwei1982@gmail.com <duanweiwei1982@gmail.com>

hiroshihorie committed:
tvOS support (#139)

17.0+ only atm

---------

Co-authored-by: cloudwebrtc <duanweiwei1982@gmail.com>

davidliu committed:
Add isDisposed to MediaStreamTrack (#140)

cloudwebrtc committed:
chore: handle invalid cipher from key size. (#142)

hiroshihorie committed:
Allow software AEC for Simulator (#143)

~Allow to use "googEchoCancellation" constraint for software AEC.
For devices "googEchoCancellation" should be false to use
VoiceProcessingIO.~

hiroshihorie committed:
Fix AudioRenderer crash & expose AVAudioPCMBuffer (#144)

cloudwebrtc committed:
fix: Fix bug for bypass voice processing. (#147)

cloudwebrtc committed:
chore: remove aes cbc for framecryptor. (#145)

hiroshihorie committed:
Change audio renderer output format (#149)

Instead of converting to Float, output original Int data without
conversion.
Output the raw format and convert when required.

hiroshihorie committed:
Fixed issue with missing network interfaces on iOS (#151)

Related issue: https://github.com/webrtc-sdk/webrtc/issues/148
Cherry-pick :
https://webrtc.googlesource.com/src/+/fea60ef8e72fb17b4f8a5363aff7e63ab8027b4f

Fixed issue with network interfaces due to a missing return value in the
"nw_path_enumerate_interfaces(...)" block. Exposed in iOS 18,
RTCNetworkMonitor::initWithObserver will only enumerate the first
interface, instead of all device interfaces

Bug: webrtc:359245764
Change-Id: Ifb9f28c33306c0096476a4afb0cdb4d734e87b2c
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/359541
Auto-Submit: Corby <corby.hoback@gmail.com>
Commit-Queue: Jonas Oreland <jonaso@webrtc.org>
Reviewed-by: Kári Helgason <kthelgason@webrtc.org>
Reviewed-by: Jonas Oreland <jonaso@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#42818}

Co-authored-by: Corby Hoback <corby.hoback@gmail.com>

davidliu committed:
Custom audio input for Android (#154)

mgsloan committed:
Use `rtc::ToString` instead of `std::to_string` in `SocketAddress::PortAsString()` (#156)

Justification for this change is that `std::to_string` should be avoided
as it uses the user's locale and calls to it get serialized, which is
bad for concurrency.

My actual motivation for this is quite bizarre. Before this change, with
Zed's use of the LiveKit Rust SDK, I was getting connection strings that
were not valid utf-8, instead having a port of
`3\u0000\u0000\u001c\u0000`. I have not figured out how that could
happen or why this change fixes it.

davidliu committed:
Fix deadlock when creating a frame cryptor on iOS (#157)

Co-authored-by: cloudwebrtc <duanweiwei1982@gmail.com>

hiroshihorie committed:
Expose apm config (#163)

davidliu committed:
Android audio prewarm (#164)

hiroshihorie committed:
Metal renderer scale patch (#165)

cloudwebrtc committed:
fix typo in framecryptor. (#167)

hiroshihorie committed:
Prefix enums with `RTC_OBJC_TYPE` macro (#171)

Continued from https://github.com/webrtc-sdk/webrtc/pull/100 also prefix
enums.
I think this shouldn't break compiling.

cloudwebrtc committed:
Bump version for boringssl to fix compiler error. (#172)

https://boringssl.googlesource.com/boringssl/+/12391e648d3c6f04a718721568cc89208b780654

pblazej committed:
Merge branch 'm125_release' into blaze/m137-patching

# Conflicts:
#	DEPS
#	api/transport/network_types.h
#	api/video_codecs/video_encoder_factory.h
#	audio/audio_state.cc
#	call/audio_state.h
#	call/rtp_transport_controller_send.cc
#	media/base/media_channel.h
#	media/engine/webrtc_video_engine.h
#	media/engine/webrtc_voice_engine.h
#	modules/audio_device/audio_device_impl.cc
#	modules/audio_device/include/audio_device.h
#	modules/audio_device/mac/audio_device_mac.cc
#	modules/audio_device/mac/audio_device_mac.h
#	modules/congestion_controller/goog_cc/goog_cc_network_control.cc
#	modules/congestion_controller/goog_cc/probe_controller.cc
#	modules/congestion_controller/goog_cc/probe_controller.h
#	modules/congestion_controller/goog_cc/probe_controller_unittest.cc
#	pc/peer_connection.cc
#	pc/video_rtp_receiver.cc
#	pc/video_rtp_receiver.h
#	sdk/BUILD.gn
#	sdk/android/BUILD.gn
#	sdk/android/api/org/webrtc/PeerConnection.java
#	sdk/android/instrumentationtests/src/org/webrtc/DefaultVideoEncoderFactoryTest.java
#	sdk/android/src/jni/pc/peer_connection.cc
#	sdk/android/src/jni/pc/peer_connection_factory.cc
#	sdk/objc/api/RTCVideoRendererAdapter.mm
#	sdk/objc/api/logging/RTCCallbackLogger.h
#	sdk/objc/api/logging/RTCCallbackLogger.mm
#	sdk/objc/api/peerconnection/RTCAudioSource+Private.h
#	sdk/objc/api/peerconnection/RTCAudioSource.mm
#	sdk/objc/api/peerconnection/RTCAudioTrack+Private.h
#	sdk/objc/api/peerconnection/RTCAudioTrack.mm
#	sdk/objc/api/peerconnection/RTCConfiguration+Private.h
#	sdk/objc/api/peerconnection/RTCConfiguration.h
#	sdk/objc/api/peerconnection/RTCConfiguration.mm
#	sdk/objc/api/peerconnection/RTCDataChannel.mm
#	sdk/objc/api/peerconnection/RTCEncodedImage+Private.mm
#	sdk/objc/api/peerconnection/RTCFieldTrials.h
#	sdk/objc/api/peerconnection/RTCFieldTrials.mm
#	sdk/objc/api/peerconnection/RTCFileLogger.h
#	sdk/objc/api/peerconnection/RTCFileLogger.mm
#	sdk/objc/api/peerconnection/RTCIceServer.h
#	sdk/objc/api/peerconnection/RTCMediaConstraints.mm
#	sdk/objc/api/peerconnection/RTCMediaSource+Private.h
#	sdk/objc/api/peerconnection/RTCMediaSource.mm
#	sdk/objc/api/peerconnection/RTCMediaStreamTrack+Private.h
#	sdk/objc/api/peerconnection/RTCMediaStreamTrack.mm
#	sdk/objc/api/peerconnection/RTCPeerConnection+Private.h
#	sdk/objc/api/peerconnection/RTCPeerConnection.h
#	sdk/objc/api/peerconnection/RTCPeerConnection.mm
#	sdk/objc/api/peerconnection/RTCPeerConnectionFactory+Native.h
#	sdk/objc/api/peerconnection/RTCPeerConnectionFactory.mm
#	sdk/objc/api/peerconnection/RTCPeerConnectionFactoryBuilder+DefaultComponents.mm
#	sdk/objc/api/peerconnection/RTCPeerConnectionFactoryBuilder.mm
#	sdk/objc/api/peerconnection/RTCRtpCodecCapability.mm
#	sdk/objc/api/peerconnection/RTCRtpCodecParameters.h
#	sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
#	sdk/objc/api/peerconnection/RTCRtpParameters.mm
#	sdk/objc/api/peerconnection/RTCRtpReceiver+Private.h
#	sdk/objc/api/peerconnection/RTCRtpReceiver.h
#	sdk/objc/api/peerconnection/RTCRtpReceiver.mm
#	sdk/objc/api/peerconnection/RTCRtpTransceiver.h
#	sdk/objc/api/peerconnection/RTCRtpTransceiver.mm
#	sdk/objc/api/peerconnection/RTCSSLAdapter.mm
#	sdk/objc/api/peerconnection/RTCSessionDescription.h
#	sdk/objc/api/peerconnection/RTCSessionDescription.mm
#	sdk/objc/api/peerconnection/RTCTracing.mm
#	sdk/objc/api/peerconnection/RTCVideoCodecInfo+Private.mm
#	sdk/objc/api/peerconnection/RTCVideoSource+Private.h
#	sdk/objc/api/peerconnection/RTCVideoSource.mm
#	sdk/objc/api/peerconnection/RTCVideoTrack.mm
#	sdk/objc/api/video_codec/RTCVideoCodecConstants.mm
#	sdk/objc/api/video_codec/RTCVideoEncoderAV1.mm
#	sdk/objc/api/video_codec/RTCVideoEncoderVP9.mm
#	sdk/objc/base/RTCLogging.h
#	sdk/objc/base/RTCLogging.mm
#	sdk/objc/base/RTCVideoCodecInfo.m
#	sdk/objc/base/RTCVideoFrame.h
#	sdk/objc/base/RTCVideoFrame.mm
#	sdk/objc/components/audio/RTCAudioSession.mm
#	sdk/objc/components/audio/RTCAudioSessionConfiguration.m
#	sdk/objc/components/audio/RTCNativeAudioSessionDelegateAdapter.h
#	sdk/objc/components/capturer/RTCCameraVideoCapturer.m
#	sdk/objc/components/capturer/RTCFileVideoCapturer.m
#	sdk/objc/components/network/RTCNetworkMonitor.mm
#	sdk/objc/components/renderer/metal/RTCMTLNSVideoView.m
#	sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.mm
#	sdk/objc/components/renderer/metal/RTCMTLRenderer.mm
#	sdk/objc/components/renderer/metal/RTCMTLVideoView.h
#	sdk/objc/components/renderer/metal/RTCMTLVideoView.m
#	sdk/objc/components/renderer/opengl/RTCEAGLVideoView.m
#	sdk/objc/components/renderer/opengl/RTCShader.mm
#	sdk/objc/components/renderer/opengl/RTCVideoViewShading.h
#	sdk/objc/components/video_codec/RTCCodecSpecificInfoH264.h
#	sdk/objc/components/video_codec/RTCDefaultVideoDecoderFactory.m
#	sdk/objc/components/video_codec/RTCDefaultVideoEncoderFactory.m
#	sdk/objc/components/video_codec/RTCH264ProfileLevelId.h
#	sdk/objc/components/video_codec/RTCH264ProfileLevelId.mm
#	sdk/objc/components/video_codec/RTCVideoDecoderFactoryH264.m
#	sdk/objc/components/video_codec/RTCVideoDecoderH264.mm
#	sdk/objc/components/video_codec/RTCVideoEncoderFactoryH264.m
#	sdk/objc/components/video_codec/RTCVideoEncoderH264.mm
#	sdk/objc/helpers/RTCCameraPreviewView.m
#	sdk/objc/helpers/RTCDispatcher+Private.h
#	sdk/objc/helpers/RTCDispatcher.h
#	sdk/objc/native/api/video_capturer.mm
#	sdk/objc/native/src/audio/audio_device_ios.mm
#	sdk/objc/native/src/audio/audio_device_module_ios.mm
#	sdk/objc/native/src/audio/voice_processing_audio_unit.mm
#	sdk/objc/native/src/objc_audio_device.mm
#	sdk/objc/native/src/objc_audio_device_delegate.h
#	sdk/objc/native/src/objc_audio_device_delegate.mm
#	sdk/objc/native/src/objc_network_monitor.h
#	sdk/objc/native/src/objc_video_frame.mm
#	sdk/objc/native/src/objc_video_track_source.h
#	sdk/objc/native/src/objc_video_track_source.mm
#	sdk/objc/unittests/RTCMTLVideoView_xctest.m
#	sdk/objc/unittests/RTCPeerConnectionFactoryBuilderTest.mm
#	sdk/objc/unittests/RTCPeerConnectionFactory_xctest.m
#	video/video_receive_stream2.cc

pblazej committed:
gn: bypass problematic visionOS check and privacy info

pblazej committed:
iOS: fix audio init flags

pblazej commented:
@cloudwebrtc @hiroshihorie @davidliu I just finished resolving the conflicts 🤷 

- The C++ part looks _reasonably good_
- I haven't checked the Android build at all - is it documented/scripted somewhere?
- Is there any automated way to [unit] test all platforms?

pblazej commented:
In some places I moved things to the deprecated namespaces, will fix that later on:

```cpp
webrtc::ArrayView<char>(name, kAdmMaxDeviceNameSize));
rtc::ArrayView<char>(name, kAdmMaxDeviceNameSize),
```

pblazej committed:
tvOS, xrOS: update build repo

pblazej committed:
Apple: build script

pblazej committed:
Apple: update build repo

pblazej committed:
Revert "gn: bypass problematic visionOS check and privacy info"

This reverts commit c6c6561757802e649a7348d77625c5eb4a5e6489.

pblazej committed:
Apple: remove duplicate (?) privacy file

pblazej committed:
Apple: add missing iOS checks

pblazej committed:
Apple: parametrize script

pblazej commented on build_xcframework_dynamic_livekit.sh:
```diff
@@ -0,0 +1,114 @@
+rm -rf $OUT_DIR/*-lib $OUT_DIR/LiveKitWebRTC.*
+
+mkdir -p $OUT_DIR/macOS-lib
+cp -R $OUT_DIR/macOS-x64/LiveKitWebRTC.framework $OUT_DIR/macOS-lib/LiveKitWebRTC.framework
```
@hiroshihorie `LiveKitWebRTC` prefix is assigned somewhere at the ninja level here?

hiroshihorie replied:
The prefixed version is in other branches, I will update those once this branch is merged 🙂
I don't think we want to commit this prefixed ver build script to main.

pblazej replied:
Yes, but it would be good to have a versioned one as well.

pblazej replied:
One more thing: does it mean that this is not working at all?
```
COMMON_ARGS="
      rtc_objc_prefix = \"LK\"
```

hiroshihorie replied:
I'm not sure since in the prefix branch, I hard code the macro in RTCMacros.h file...

davidliu commented:
For android builds, we have build commands here: https://github.com/webrtc-sdk/webrtc-build/blob/main/docs/build.md

You can kick off a build manually through github actions [here](https://github.com/webrtc-sdk/webrtc-build/actions/workflows/build.yml) and passing in a commit hash to the workflow dispatch. I started building one off of 509ebe4. The prefixed build will probably not work since it's code-sensitive and usually requires extra manual changes separate from merging with upstream.

pblazej commented on modules/congestion_controller/goog_cc/probe_controller.cc:
```diff
@@ -286,21 +287,6 @@ std::vector<ProbeClusterConfig> ProbeController::OnNetworkAvailability(
   return std::vector<ProbeClusterConfig>();
 }
 
-void ProbeController::UpdateState(State new_state) {
```
Introduced in 8505a9838ea91c66c96c173d30cd66f9dbcc7548 (merged later into m125), not sure if applicable.

pblazej commented on sdk/objc/api/peerconnection/RTCPeerConnectionFactory.mm:
This one generated the most conflicts, would be nice to take a deeper look at the initializer hierarchy here.

pblazej commented on sdk/objc/native/src/audio/audio_device_ios.mm:
```diff
@@ -843,7 +903,7 @@ static void LogDeviceInfo() {
   // If we're not initialized we don't need to do anything. Audio unit will
   // be initialized on initialization.
-  if (!audio_is_initialized_) return;
+  if (!playout_is_initialized_ && !recording_is_initialized_) return;
```
Here and elsewhere in this file: check if this logic is still correct.

pblazej committed:
CR: Revert some namespaces

pblazej committed:
ObjC: Fix init warnings

pblazej committed:
xrOS: bump target

pblazej commented on build_xcframework_dynamic_livekit.sh:
We can move it to the https://github.com/webrtc-sdk/webrtc-build, wdyt @hiroshihorie?

pblazej commented:
@davidliu looks like Android is failing on these patches, started fixing [them](https://github.com/webrtc-sdk/webrtc-build/pull/32), not sure if the [build](https://github.com/webrtc-sdk/webrtc-build/actions/runs/15733111237/job/44338759255) succeeds.

pblazej commented:
@cloudwebrtc any possibility to test other platforms with/without Docker on a purely macOS machine (CI is not enough I guess, especially the windows stuff commented out etc.)?

cloudwebrtc commented:
Apple Silicon macOS cannot test windows/linux code under X86 arch through VMware (only arm64). In the past, I compiled a libwebrtc.dll to upgrade flutter-webrtc after the patch was completed, and did a functional check of webrtc on e2e-flutter (on a x64 windows pc). But device-related tests, such as switching audio input/output, camera acquisition resolution fps, etc., may still require some manual testing.

cloudwebrtc commented:
In the previous versions (m104, m114, m125), I did two rounds of cherry-picking PRs. The first one was similar to what you are doing, merging the big commit of the previous PR and the PRs added later. The second cherry-pick will classify the PRs added later and compress them into larger commits, so as to avoid too many commits accumulating and causing tons of work in the next upgrade.

cloudwebrtc commented:
Since conflicts between similar versions are much smaller than those between versions with a larger span, compressing them into a large commit usually does not cause too many conflicts in the next upgrade (the workload of resolving conflicts will be within an acceptable range)

pblazej commented:
> In the previous versions (m104, m114, m125), I did two rounds of cherry-picking PRs. The first one was similar to what you are doing, merging the big commit of the previous PR and the PRs added later. The second cherry-pick will classify the PRs added later and compress them into larger commits, so as to avoid too many commits accumulating and causing tons of work in the next upgrade.

Yes, there was no clear pattern in these "later" commits, I assumed that we can repeat the procedure - merge old→new and resolve m137+all_the_updates in one go as I did here.

pblazej committed:
Android: silence deprecation

pblazej committed:
Android: fix optional

pblazej committed:
Android: fix more build errors

pblazej committed:
Android: fix imports

pblazej committed:
Android: fix jni

pblazej committed:
Android: fix ordinal warning

pblazej commented:
@davidliu I managed to get Android 🟢 https://github.com/webrtc-sdk/webrtc-build/actions/runs/15818791300/job/44582957348

Fixes in webrtc-build: https://github.com/webrtc-sdk/webrtc-build/pull/32

Not sure if all of them are idiomatic ⬆️, let me know if there are any further steps from the Android point of view (especially how to run those patched tests) 👍 

Is it `webrtc-build/build/run.py build android --test`?

@cloudwebrtc assuming arm64, discarding Docker for now, how do you normally run C++/Obj-C++/Android side tests here to check the integrity? I believe we can postpone Flutter for now.

In other words, what checks should I run before attempting to pass this one for the final review?

pblazej committed:
Build: update pin

hiroshihorie committed:
Fix prefixing

hiroshihorie committed:
Remove duplicate code block in RTCCameraVideoCapturer

hiroshihorie committed:
Extra check when removing capture connection

Co-Authored-By: Błażej Pankowski <86720177+pblazej@users.noreply.github.com>

davidliu reviewed (approved):
Ran through E2E tests on android and seemed to work fine.

hiroshihorie committed:
m137 AVAudioEngine ver ADM (#178)

Co-authored-by: Błażej Pankowski <86720177+pblazej@users.noreply.github.com>

pblazej committed:
Apple: remove build script/moved to build repo

cloudwebrtc commented:
The patch looks good. I will try to do a full platform e2e test on the flutter sdk this week. There should be results soon.

hiroshihorie commented:
@cloudwebrtc I think we will need to test on this branch: https://github.com/flutter-webrtc/flutter-webrtc/pull/1875 because of audio changes etc. I will publish a new Pod and update the PR.

cloudwebrtc commented:
All platforms seem to work fine except Android.
There is a crash due to webrtc_voice_engine.cc update https://github.com/webrtc-sdk/webrtc/pull/173/files#diff-56f5e0c459b287281ef3b0431d3f4129e8e4be4c6955d845bcb22210f08b7ba5R1755

`adm->SetMicrophoneMute(is_all_muted);`

Created a fix here.

https://github.com/webrtc-sdk/webrtc/pull/180

pblazej commented:
Cooool, I'm happy to merge this one as soon as Android is resolved 🫴

cloudwebrtc committed:
fix: avoid rtc check notreached assertion crashes (#180)

crash from this line.

https://github.com/webrtc-sdk/webrtc/pull/173/files#diff-56f5e0c459b287281ef3b0431d3f4129e8e4be4c6955d845bcb22210f08b7ba5R1755


https://github.com/webrtc-sdk/webrtc/blob/blaze/m137-patching/sdk/android/src/jni/audio_device/audio_device_module.cc#L433

<img width="2136" height="532"
alt="a8fd508a-7b95-4f29-9f7b-0224ca77bf07"
src="https://github.com/user-attachments/assets/0b15f0e6-83af-40e5-9296-ffaf03b4be43"
/>

pblazej commented:
Latest build with Android fix https://github.com/webrtc-sdk/webrtc-build/actions/runs/16412378269

cloudwebrtc reviewed (approved):
lgtm!
